Source/WebCore/page/EventHandler.cpp

```diff
@@ -1017,9 +1017,19 @@ void EventHandler::updateSelectionForMouseDrag(const HitTestResult& hitTestResul
             newSelection.setExtent(targetPosition);
     }
 
-    if (m_frame.selection().granularity() != TextGranularity::CharacterGranularity)
+    if (m_frame.selection().granularity() != TextGranularity::CharacterGranularity) {
         newSelection.expandUsingGranularity(m_frame.selection().granularity());
 
+        auto dragStartPosition = m_mousePressNode->renderer()->positionForPoint(m_dragStartPosition, nullptr);
+        VisibleSelection dragStartSelection = dragStartPosition;
+        dragStartSelection.expandUsingGranularity(m_frame.selection().granularity());
```

Comment:
Actually, we don't need this selection. We can just use oldSelection instead.

Reply:
Hm... we'd need to remember the original selection instead.
There is some kind of a bug in positionForPoint...